scripts/Test-FaultInjection.ps1

```diff
@@ -22,6 +22,16 @@ Set-Content -Path ($TestProgram + ".fault.log") ""
 
 $iteration = 0
 
+$timer = New-Object System.Timers.Timer
+# Set timer for 5 minutes
+$timer.Interval = 300000
+
+# When the watchdog timer fires, kill the test binary.
+Register-ObjectEvent -InputObject $timer -EventName Elapsed -Action {
+    Write-Error "Test binary exceede time"
```

Comment:
Correct spelling: exceeded.

Also, will the user know which binary exceeded the time to cancel?

Reply:
This should gather a memory dump of the hung process, but that requires some additional work. The first version is to improve the reliability of the CI/CD pipeline.

Reply:
Can we have an issue tracking the improvement for the next step?